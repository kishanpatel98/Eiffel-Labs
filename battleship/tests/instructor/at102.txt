-- user test 2
undo
redo
undo
redo
undo
redo
debug_test(medium)
give_up
debug_test(medium)
undo
redo
custom_setup_test(6,3,30,3)
custom_setup_test(6,1,30,3)
undo
undo
undo
redo
redo
redo
fire([C,3])
bomb([D,2],[D,3])
fire([C,2])
fire([A,2])
bomb([C,4],[B,4])
fire([E,2])
undo
redo
undo
redo
give_up
new_game(easy)
give_up
custom_setup_test(6,4,30,3)
bomb([C,4],[C,2])
bomb([C,4],[C,3])
bomb([C,3],[C,1])
undo
undo
fire([C,4])
bomb([B,3],[B,4])
undo
redo
redo
fire([A,2])
