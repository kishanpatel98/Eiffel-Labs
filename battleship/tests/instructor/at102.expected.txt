  state 0 OK -> Start a new game
->undo
  state 1 Nothing to undo -> Start a new game
->redo
  state 2 Nothing to redo -> Start a new game
->undo
  state 3 Nothing to undo -> Start a new game
->redo
  state 4 Nothing to redo -> Start a new game
->undo
  state 5 Nothing to undo -> Start a new game
->redo
  state 6 Nothing to redo -> Start a new game
->debug_test(medium)
  state 7 OK -> Fire Away!
     1  2  3  4  5  6
  A  _  _  _  _  _  _
  B  _  _  _  _  _  _
  C  _  h  h  _  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 0/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->give_up
  state 8 OK -> You gave up!
     1  2  3  4  5  6
  A  _  _  _  _  _  _
  B  _  _  _  _  _  _
  C  _  h  h  _  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 0/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->debug_test(medium)
  state 9 OK -> Fire Away!
     1  2  3  4  5  6
  A  _  h  _  _  _  _
  B  _  _  _  v  _  _
  C  _  v  _  v  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 0/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [C, 2]->v;[D, 2]->v;[E, 2]->v
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [A, 2]->h
->undo
  state 10 Nothing to undo -> Fire Away!
     1  2  3  4  5  6
  A  _  h  _  _  _  _
  B  _  _  _  v  _  _
  C  _  v  _  v  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 0/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [C, 2]->v;[D, 2]->v;[E, 2]->v
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [A, 2]->h
->redo
  state 11 Nothing to redo -> Fire Away!
     1  2  3  4  5  6
  A  _  h  _  _  _  _
  B  _  _  _  v  _  _
  C  _  v  _  v  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 0/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [C, 2]->v;[D, 2]->v;[E, 2]->v
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [A, 2]->h
->custom_setup_test(6,3,30,3)
  state 12 Game already started -> Fire Away!
     1  2  3  4  5  6
  A  _  h  _  _  _  _
  B  _  _  _  v  _  _
  C  _  v  _  v  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 0/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [C, 2]->v;[D, 2]->v;[E, 2]->v
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [A, 2]->h
->custom_setup_test(6,1,30,3)
  state 13 Game already started -> Fire Away!
     1  2  3  4  5  6
  A  _  h  _  _  _  _
  B  _  _  _  v  _  _
  C  _  v  _  v  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 0/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [C, 2]->v;[D, 2]->v;[E, 2]->v
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [A, 2]->h
->undo
  state 14 (= state 12) Game already started -> Fire Away!
     1  2  3  4  5  6
  A  _  h  _  _  _  _
  B  _  _  _  v  _  _
  C  _  v  _  v  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 0/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [C, 2]->v;[D, 2]->v;[E, 2]->v
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [A, 2]->h
->undo
  state 15 (= state 9) OK -> Fire Away!
     1  2  3  4  5  6
  A  _  h  _  _  _  _
  B  _  _  _  v  _  _
  C  _  v  _  v  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 0/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [C, 2]->v;[D, 2]->v;[E, 2]->v
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [A, 2]->h
->undo
  state 16 Nothing to undo -> Fire Away!
     1  2  3  4  5  6
  A  _  h  _  _  _  _
  B  _  _  _  v  _  _
  C  _  v  _  v  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 0/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [C, 2]->v;[D, 2]->v;[E, 2]->v
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [A, 2]->h
->redo
  state 17 (= state 12) Game already started -> Fire Away!
     1  2  3  4  5  6
  A  _  h  _  _  _  _
  B  _  _  _  v  _  _
  C  _  v  _  v  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 0/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [C, 2]->v;[D, 2]->v;[E, 2]->v
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [A, 2]->h
->redo
  state 18 (= state 13) Game already started -> Fire Away!
     1  2  3  4  5  6
  A  _  h  _  _  _  _
  B  _  _  _  v  _  _
  C  _  v  _  v  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 0/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [C, 2]->v;[D, 2]->v;[E, 2]->v
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [A, 2]->h
->redo
  state 19 Nothing to redo -> Fire Away!
     1  2  3  4  5  6
  A  _  h  _  _  _  _
  B  _  _  _  v  _  _
  C  _  v  _  v  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 0/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [C, 2]->v;[D, 2]->v;[E, 2]->v
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [A, 2]->h
->fire([C, 3])
  state 20 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  _  h  _  _  _  _
  B  _  _  _  v  _  _
  C  _  v  O  v  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 1/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [C, 2]->v;[D, 2]->v;[E, 2]->v
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [A, 2]->h
->bomb([D, 2],[D, 3])
  state 21 OK -> Hit! Keep Firing!
     1  2  3  4  5  6
  A  _  h  _  _  _  _
  B  _  _  _  v  _  _
  C  _  v  O  v  _  _
  D  _  X  O  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 1/16
  Bombs: 1/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [C, 2]->v;[D, 2]->X;[E, 2]->v
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [A, 2]->h
->fire([C, 2])
  state 22 OK -> Hit! Keep Firing!
     1  2  3  4  5  6
  A  _  h  _  _  _  _
  B  _  _  _  v  _  _
  C  _  X  O  v  _  _
  D  _  X  O  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 2/16
  Bombs: 1/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [C, 2]->X;[D, 2]->X;[E, 2]->v
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [A, 2]->h
->fire([A, 2])
  state 23 OK -> 1x1 ship sunk! Keep Firing!
     1  2  3  4  5  6
  A  _  X  _  _  _  _
  B  _  _  _  v  _  _
  C  _  X  O  v  _  _
  D  _  X  O  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 3/16
  Bombs: 1/3
  Score: 1/6 (Total: 1/6)
  Ships: 1/3
    3x1: [C, 2]->X;[D, 2]->X;[E, 2]->v
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [A, 2]->X
->bomb([C, 4],[B, 4])
  state 24 OK -> 2x1 ship sunk! Keep Firing!
     1  2  3  4  5  6
  A  _  X  _  _  _  _
  B  _  _  _  X  _  _
  C  _  X  O  X  _  _
  D  _  X  O  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 3/16
  Bombs: 2/3
  Score: 3/6 (Total: 3/6)
  Ships: 2/3
    3x1: [C, 2]->X;[D, 2]->X;[E, 2]->v
    2x1: [B, 4]->X;[C, 4]->X
    1x1: [A, 2]->X
->fire([E, 2])
  state 25 OK -> 3x1 ship sunk! You Win!
     1  2  3  4  5  6
  A  _  X  _  _  _  _
  B  _  _  _  X  _  _
  C  _  X  O  X  _  _
  D  _  X  O  _  _  _
  E  _  X  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 4/16
  Bombs: 2/3
  Score: 6/6 (Total: 6/6)
  Ships: 3/3
    3x1: [C, 2]->X;[D, 2]->X;[E, 2]->X
    2x1: [B, 4]->X;[C, 4]->X
    1x1: [A, 2]->X
->undo
  state 26 Nothing to undo -> Start a new game
     1  2  3  4  5  6
  A  _  X  _  _  _  _
  B  _  _  _  X  _  _
  C  _  X  O  X  _  _
  D  _  X  O  _  _  _
  E  _  X  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 4/16
  Bombs: 2/3
  Score: 6/6 (Total: 6/6)
  Ships: 3/3
    3x1: [C, 2]->X;[D, 2]->X;[E, 2]->X
    2x1: [B, 4]->X;[C, 4]->X
    1x1: [A, 2]->X
->redo
  state 27 Nothing to redo -> Start a new game
     1  2  3  4  5  6
  A  _  X  _  _  _  _
  B  _  _  _  X  _  _
  C  _  X  O  X  _  _
  D  _  X  O  _  _  _
  E  _  X  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 4/16
  Bombs: 2/3
  Score: 6/6 (Total: 6/6)
  Ships: 3/3
    3x1: [C, 2]->X;[D, 2]->X;[E, 2]->X
    2x1: [B, 4]->X;[C, 4]->X
    1x1: [A, 2]->X
->undo
  state 28 Nothing to undo -> Start a new game
     1  2  3  4  5  6
  A  _  X  _  _  _  _
  B  _  _  _  X  _  _
  C  _  X  O  X  _  _
  D  _  X  O  _  _  _
  E  _  X  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 4/16
  Bombs: 2/3
  Score: 6/6 (Total: 6/6)
  Ships: 3/3
    3x1: [C, 2]->X;[D, 2]->X;[E, 2]->X
    2x1: [B, 4]->X;[C, 4]->X
    1x1: [A, 2]->X
->redo
  state 29 Nothing to redo -> Start a new game
     1  2  3  4  5  6
  A  _  X  _  _  _  _
  B  _  _  _  X  _  _
  C  _  X  O  X  _  _
  D  _  X  O  _  _  _
  E  _  X  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 4/16
  Bombs: 2/3
  Score: 6/6 (Total: 6/6)
  Ships: 3/3
    3x1: [C, 2]->X;[D, 2]->X;[E, 2]->X
    2x1: [B, 4]->X;[C, 4]->X
    1x1: [A, 2]->X
->give_up
  state 30 Game not started -> Start a new game
     1  2  3  4  5  6
  A  _  X  _  _  _  _
  B  _  _  _  X  _  _
  C  _  X  O  X  _  _
  D  _  X  O  _  _  _
  E  _  X  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 4/16
  Bombs: 2/3
  Score: 6/6 (Total: 6/6)
  Ships: 3/3
    3x1: [C, 2]->X;[D, 2]->X;[E, 2]->X
    2x1: [B, 4]->X;[C, 4]->X
    1x1: [A, 2]->X
->new_game(easy)
  state 31 OK -> Fire Away!
     1  2  3  4
  A  _  _  _  _
  B  _  _  _  _
  C  _  _  _  _
  D  _  _  _  _
  Current Game: 1
  Shots: 0/8
  Bombs: 0/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: Not Sunk
    1x1: Not Sunk
->give_up
  state 32 OK -> You gave up!
     1  2  3  4
  A  _  _  _  _
  B  _  _  _  _
  C  _  _  _  _
  D  _  _  _  _
  Current Game: 1
  Shots: 0/8
  Bombs: 0/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: Not Sunk
    1x1: Not Sunk
->custom_setup_test(6,4,30,3)
  state 33 OK -> Fire Away!
     1  2  3  4  5  6
  A  _  h  h  _  _  _
  B  _  v  _  _  _  _
  C  _  v  h  h  h  v
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 0/30
  Bombs: 0/3
  Score: 0/10 (Total: 0/10)
  Ships: 0/4
    4x1: [B, 2]->v;[C, 2]->v;[D, 2]->v;[E, 2]->v
    3x1: [C, 3]->h;[C, 4]->h;[C, 5]->h
    2x1: [A, 2]->h;[A, 3]->h
    1x1: [C, 6]->v
->bomb([C, 4],[C, 2])
  state 34 Bomb coordinates must be adjacent -> Fire Away!
     1  2  3  4  5  6
  A  _  h  h  _  _  _
  B  _  v  _  _  _  _
  C  _  v  h  h  h  v
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 0/30
  Bombs: 0/3
  Score: 0/10 (Total: 0/10)
  Ships: 0/4
    4x1: [B, 2]->v;[C, 2]->v;[D, 2]->v;[E, 2]->v
    3x1: [C, 3]->h;[C, 4]->h;[C, 5]->h
    2x1: [A, 2]->h;[A, 3]->h
    1x1: [C, 6]->v
->bomb([C, 4],[C, 3])
  state 35 OK -> Hit! Keep Firing!
     1  2  3  4  5  6
  A  _  h  h  _  _  _
  B  _  v  _  _  _  _
  C  _  v  X  X  h  v
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 0/30
  Bombs: 1/3
  Score: 0/10 (Total: 0/10)
  Ships: 0/4
    4x1: [B, 2]->v;[C, 2]->v;[D, 2]->v;[E, 2]->v
    3x1: [C, 3]->X;[C, 4]->X;[C, 5]->h
    2x1: [A, 2]->h;[A, 3]->h
    1x1: [C, 6]->v
->bomb([C, 3],[C, 1])
  state 36 Bomb coordinates must be adjacent -> Keep Firing!
     1  2  3  4  5  6
  A  _  h  h  _  _  _
  B  _  v  _  _  _  _
  C  _  v  X  X  h  v
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 0/30
  Bombs: 1/3
  Score: 0/10 (Total: 0/10)
  Ships: 0/4
    4x1: [B, 2]->v;[C, 2]->v;[D, 2]->v;[E, 2]->v
    3x1: [C, 3]->X;[C, 4]->X;[C, 5]->h
    2x1: [A, 2]->h;[A, 3]->h
    1x1: [C, 6]->v
->undo
  state 37 (= state 35) OK -> Hit! Keep Firing!
     1  2  3  4  5  6
  A  _  h  h  _  _  _
  B  _  v  _  _  _  _
  C  _  v  X  X  h  v
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 0/30
  Bombs: 1/3
  Score: 0/10 (Total: 0/10)
  Ships: 0/4
    4x1: [B, 2]->v;[C, 2]->v;[D, 2]->v;[E, 2]->v
    3x1: [C, 3]->X;[C, 4]->X;[C, 5]->h
    2x1: [A, 2]->h;[A, 3]->h
    1x1: [C, 6]->v
->undo
  state 38 (= state 34) Bomb coordinates must be adjacent -> Fire Away!
     1  2  3  4  5  6
  A  _  h  h  _  _  _
  B  _  v  _  _  _  _
  C  _  v  h  h  h  v
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 0/30
  Bombs: 0/3
  Score: 0/10 (Total: 0/10)
  Ships: 0/4
    4x1: [B, 2]->v;[C, 2]->v;[D, 2]->v;[E, 2]->v
    3x1: [C, 3]->h;[C, 4]->h;[C, 5]->h
    2x1: [A, 2]->h;[A, 3]->h
    1x1: [C, 6]->v
->fire([C, 4])
  state 39 OK -> Hit! Keep Firing!
     1  2  3  4  5  6
  A  _  h  h  _  _  _
  B  _  v  _  _  _  _
  C  _  v  h  X  h  v
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 1/30
  Bombs: 0/3
  Score: 0/10 (Total: 0/10)
  Ships: 0/4
    4x1: [B, 2]->v;[C, 2]->v;[D, 2]->v;[E, 2]->v
    3x1: [C, 3]->h;[C, 4]->X;[C, 5]->h
    2x1: [A, 2]->h;[A, 3]->h
    1x1: [C, 6]->v
->bomb([B, 3],[B, 4])
  state 40 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  _  h  h  _  _  _
  B  _  v  O  O  _  _
  C  _  v  h  X  h  v
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 1/30
  Bombs: 1/3
  Score: 0/10 (Total: 0/10)
  Ships: 0/4
    4x1: [B, 2]->v;[C, 2]->v;[D, 2]->v;[E, 2]->v
    3x1: [C, 3]->h;[C, 4]->X;[C, 5]->h
    2x1: [A, 2]->h;[A, 3]->h
    1x1: [C, 6]->v
->undo
  state 41 (= state 39) OK -> Hit! Keep Firing!
     1  2  3  4  5  6
  A  _  h  h  _  _  _
  B  _  v  _  _  _  _
  C  _  v  h  X  h  v
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 1/30
  Bombs: 0/3
  Score: 0/10 (Total: 0/10)
  Ships: 0/4
    4x1: [B, 2]->v;[C, 2]->v;[D, 2]->v;[E, 2]->v
    3x1: [C, 3]->h;[C, 4]->X;[C, 5]->h
    2x1: [A, 2]->h;[A, 3]->h
    1x1: [C, 6]->v
->redo
  state 42 (= state 40) OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  _  h  h  _  _  _
  B  _  v  O  O  _  _
  C  _  v  h  X  h  v
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 1/30
  Bombs: 1/3
  Score: 0/10 (Total: 0/10)
  Ships: 0/4
    4x1: [B, 2]->v;[C, 2]->v;[D, 2]->v;[E, 2]->v
    3x1: [C, 3]->h;[C, 4]->X;[C, 5]->h
    2x1: [A, 2]->h;[A, 3]->h
    1x1: [C, 6]->v
->redo
  state 43 Nothing to redo -> Keep Firing!
     1  2  3  4  5  6
  A  _  h  h  _  _  _
  B  _  v  O  O  _  _
  C  _  v  h  X  h  v
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 1/30
  Bombs: 1/3
  Score: 0/10 (Total: 0/10)
  Ships: 0/4
    4x1: [B, 2]->v;[C, 2]->v;[D, 2]->v;[E, 2]->v
    3x1: [C, 3]->h;[C, 4]->X;[C, 5]->h
    2x1: [A, 2]->h;[A, 3]->h
    1x1: [C, 6]->v
->fire([A, 2])
  state 44 OK -> Hit! Keep Firing!
     1  2  3  4  5  6
  A  _  X  h  _  _  _
  B  _  v  O  O  _  _
  C  _  v  h  X  h  v
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  _  _  _  _  _
  Current Game (debug): 1
  Shots: 2/30
  Bombs: 1/3
  Score: 0/10 (Total: 0/10)
  Ships: 0/4
    4x1: [B, 2]->v;[C, 2]->v;[D, 2]->v;[E, 2]->v
    3x1: [C, 3]->h;[C, 4]->X;[C, 5]->h
    2x1: [A, 2]->X;[A, 3]->h
    1x1: [C, 6]->v
